stack/adding-a-new-package.rst

```diff
@@ -35,29 +45,45 @@ Replace ``afw`` with the relevant package name to get to the correct page on Git
-With this enabled people will be able to use the GitHub merge button on Pull Requests and know that the :ref:`standard process <workflow-code-review-merge>` is being adhered to.
+For ``afw`` this is located at https://github.com/lsst/afw/settings/branches/ and can also be found from the "Branches" sidebar item on the settings screen.
+In the "Branch protection rules" section of that page you will have to click on "Add rule" to create a rule for ``master``.
+Add in ``master`` as the branch name pattern and then enable status checks and require that branches be up to date before merging.
+Administrators must be included in these protections since it's all too easy to make a mistake without realizing you have special override powers.
+With checks enabled people will be able to use the GitHub merge button on Pull Requests and know that the :ref:`standard process <workflow-code-review-merge>` is being adhered to.
 
 .. image:: /_static/build-ci/github_branch_protection.png
 
-In the image above no automated status checks are being performed.
-Whilst not required, we recommend that some basic checks are enabled on repositories using Travis.
-This will not replace normal testing with :doc:`Jenkins job <jenkins-stack-os-matrix>` but for packages that have been updated to :ref:`use flake8 <testing-flake8>` it is useful to add a simple Travis script like the following:
+In the image above no automated status checks are being performed because we have not yet enabled any.
+GitHub requires that at least one check runs before the up-to-date checks are enabled so a Travis job **must** be provided if the GitHub merge button is to be used.
+Travis does not replace normal testing done with a :doc:`Jenkins job <jenkins-stack-os-matrix>`, but for packages that have been updated to :ref:`use flake8 <testing-flake8>` it is useful to add a simple Travis script like the following:
 
 .. literalinclude:: examples/flake8-travis.yml
   :language: yaml
 
+.. note::
+
+   For Travis to recognize the file in the repository it **must** be named :file:`.travis.yml` and be in the repository root.
+   Any other name will be ignored and in particular the common misnaming of the file as :file:`.travis.yaml` does not work.
+
 For packages containing C++ that have been tidied up using ``clang`` tools, you may consider adding a Travis check that runs the tidy tool and does a ``diff`` with the repository.
+Shell scripts can also be checked by calling the ``shellcheck`` command.
+If nothing seems appropriate a null Travis job should be enabled to allow GitHub to do the checks it needs.
+An example null Travis file follows:
+
+.. literalinclude:: examples/null-travis.yml
+  :language: yaml
+
+Once the Travis script has been written and pushed to GitHub, new pull requests will automatically run the Travis checks without any further configuration and the results of the Travis checks will be visible in the "Checks" tab of the pull request on GitHub.
```

Comment:
Shouldn't we insist that the script be merged to master, not just pushed to GitHub?
```suggestion
Once the Travis script has been written, pushed to GitHub, and merged to master, new pull requests will automatically run the Travis checks without any further configuration and the results of the Travis checks will be visible in the "Checks" tab of the pull request on GitHub.
```

Reply:
Appearing on the branch is enough for that branch to enable travis. I couldnt' imagine people not merging it though 😄